packages/react/toolbar/src/Toolbar.tsx

```diff
@@ -0,0 +1,246 @@
+ * ToolbarLabel
+ * -----------------------------------------------------------------------------------------------*/
+
+const LABEL_NAME = 'ToolbarLabel';
+
+type ToolbarLabelOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarLabelPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarLabelOwnProps
+>;
+
+const ToolbarLabel = React.forwardRef((props, forwardedRef) => (
+  <Primitive selector={getSelector(LABEL_NAME)} {...props} ref={forwardedRef} />
+)) as ToolbarLabelPrimitive;
+
+ToolbarLabel.displayName = LABEL_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarSeparator
+ * -----------------------------------------------------------------------------------------------*/
+
+const SEPARATOR_NAME = 'ToolbarSeparator';
+
+type ToolbarSeparatorOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarSeparatorPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarSeparatorOwnProps
+>;
+
+const ToolbarSeparator = React.forwardRef((props, forwardedRef) => {
+  const { orientation } = useToolbarContext(GROUP_NAME);
+
+  return (
+    <Primitive
+      role="separator"
+      selector={getSelector(SEPARATOR_NAME)}
+      aria-orientation={orientation}
+      data-orientation={orientation}
+      {...props}
+      ref={forwardedRef}
+    />
+  );
+}) as ToolbarSeparatorPrimitive;
+
+ToolbarSeparator.displayName = SEPARATOR_NAME;
+
+/* -------------------------------------------------------------------------------------------------
+ * ToolbarItem
+ * -----------------------------------------------------------------------------------------------*/
+
+const ITEM_NAME = 'ToolbarItem';
+
+type ToolbarItemOwnProps = Polymorphic.OwnProps<typeof Primitive>;
+type ToolbarItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof Primitive>,
+  ToolbarItemOwnProps
+>;
+
+const ToolbarItem = React.forwardRef((props, forwardedRef) => {
+  const { disabled, ...itemProps } = props as any;
+
+  const rovingFocusProps = useRovingFocus({
+    disabled: disabled,
+    active: false,
+  });
+
+  const handleClick = (event: MouseEvent) => {
+    if (disabled) {
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <Primitive
+      ref={forwardedRef}
+      {...itemProps}
+      {...rovingFocusProps}
+      role="toolbaritem"
+      selector={getSelector(ITEM_NAME)}
+      aria-disabled={disabled}
+      data-disabled={disabled}
+      onClick={handleClick}
+      // RovingFocus eventHandlers should not block item eventHandlers
+      onFocus={composeEventHandlers(itemProps.onFocus, rovingFocusProps.onFocus)}
+      onKeyDown={composeEventHandlers(itemProps.onKeyDown, rovingFocusProps.onKeyDown)}
+      onMouseDown={composeEventHandlers(itemProps.onMouseDown, rovingFocusProps.onMouseDown)}
```

Comment:
When our handlers are small (like your `onClick` one) we usually declare them inline with `composeEventHandlers` which gives the added bonus of having the `event` param already typed for you:

```suggestion
      // RovingFocus eventHandlers should not block item eventHandlers
      onFocus={composeEventHandlers(itemProps.onFocus, rovingFocusProps.onFocus)}
      onKeyDown={composeEventHandlers(itemProps.onKeyDown, rovingFocusProps.onKeyDown)}
      onMouseDown={composeEventHandlers(itemProps.onMouseDown, rovingFocusProps.onMouseDown)}
      onClick={composeEventHandlers(itemProps.onClick, event => {
        if (disabled) {
          event.preventDefault();
        }
      })}
```

Reply:
Another note on that one, I don't believe `rovingProps` contain `onMouseDown`, so that line should be able to go also.

Reply:
Actually ignore that, you are 💯 right! I missed the early return (when item is disabled) when looking at our source 👍

Reply:
@jjenzz Shouldn't be the other way around? 
```js
 onClick={composeEventHandlers(
     (event) => disabled && event.preventDefault(),
     itemProps.onClick
)}
```

Isn't the intended behavior is to prevent the execution of the `itemProps.onClick` if the item is disabled even if the HTML element isn't `button`?

Reply:
Or maybe this is abusing the `composeEventHndlers` so something like this instead?

```ts
      onClick={(event) => (disabled ? event.preventDefault() : itemProps.onClick?.(event))}
```

Reply:
Ah yes, sorry! If we don't want the consumer's event to fire then your second example is correct :+1:

Reply:
I'm a bit confused on that one, with `composeEventHandlers` we generally do this
```tsx
composeEventHandlers(props.eventHandler, () => { /*... our internal handler code */ })
```

After all, this is its signature:

![image](https://user-images.githubusercontent.com/1539897/107284696-165f2d00-6a56-11eb-81e4-4ffd7a1e4ccb.png)

I'm probably missing something here though… 😅

Reply:
@benoitgrelard the way it is implemented allows the `originalEventHandler` to preventDefault and thus prevent `ourEventHandler` unless `checkForDefultPrevented` is set to false.
```ts
export function composeEventHandlers<E extends React.SyntheticEvent | Event>(
  originalEventHandler?: (event: E) => void,
  ourEventHandler?: (event: E) => void,
  { checkForDefaultPrevented = true } = {}
) {
  return function handleEvent(event: E) {
    originalEventHandler?.(event);

    if (checkForDefaultPrevented === false || !event.defaultPrevented) {
      return ourEventHandler?.(event);
    }
  };
}
```

Now I want to achieve the opposite, I want to preventDefault and skip the `itemProps.onClick()` if the item is `disabled` without relying on the normal behavior of `button` for example. so initially I thought about abusing `composeEventHandlers` by switching the order of event handlers, but then thought it is better to do it manually like this:
```ts
 onClick={(event) => (disabled ? event.preventDefault() : itemProps.onClick?.(event))}
```

And because `rovingFocusGroup` doesn't have `onClick` and we don't need to add any other handlers `onClick`, so `composeEventHandlers` is irrelevant here anymore, the confusion comes from my initial thought to use it anyway out of laziness :D sorry.

Reply:
Looking closer on that one, I think it doesn't really make sense to even have the `onClick` at all. From a component encapsulation perspective, the toolbar and toolbar item are basically just managing a group of control and roving focus. From that standpoint, roving focus would skip the item if it's disabled. Whether the component that is passed through the `as` prop supports being disabled as a control is really it's own concern, not the toolbar item's concern.